src/bindings/writeTypescript.ts

```diff
@@ -272,58 +303,66 @@ export function writeTypescript(
 
         if (init) {
             w.append(
-                `static async init(${writeArguments(init.args).join(", ")}) {`,
+                `static async init<T extends ${abi.name}>(` +
+                    `this: new (address: Address, init?: { code: Cell, data: Cell }) => T,` +
```

Comment:
This kind of types is meant to type legacy JS code, not to write or generate novel code.